Implement shop 1

Aichnerc opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently, there is no shop implemented. #31 
- Currently, the footer does not fit the theme. #27 
- Currently, the website has countless little fields of interest which require improvement.


**What is the new behavior (if this is a feature change)?**
- The shop has been implemented. #31 
- The footer has been improved. #27 
- The Sub and Blackwolf components have been updated.
- The FAQ has been updated.
- The navigation icon is now the button to launch the shop cart.
- Types in the dummy data have been fixed.
- The code quality has been improved by enforcing our guidelines.

**Other information**:
- The version has been updated to v0.3.0
[Learn more about our versioning](https://github.com/aichner/React-MDB-Template#versioning)

Aichnerc committed:
Add required modules

New modules have been added.

Aichnerc committed:
Add queries

The queries have been added.

Aichnerc committed:
Remove package-lock

The package-lock file has been excluded from git.

Aichnerc committed:
Add .env

The .env has been added.

Aichnerc committed:
Add basic connection

The basic apollo react setup has been established.

Aichnerc committed:
Add react-apollo

A missing package has been added.

Aichnerc committed:
Update gitignore

The package-lock has been excluded for good.

Aichnerc committed:
Add ApolloProvider

The apollo provider has been added.

Aichnerc committed:
Add modules

Some missing modules have been added.

Aichnerc committed:
Add LineItem

The line item atom has been added.

Aichnerc committed:
Add Cart

The cart molecule has been added.

Aichnerc committed:
Add connection

The HomePage has been adjusted to be able to
receive data.

Aichnerc committed:
Improve code quality

The code quality has been improved.

Aichnerc committed:
Add Product

A product molecule has been added.

Aichnerc committed:
Add params

Parameters have been added to the Shop component.

Aichnerc committed:
Improve code quality

CSS Code formatting.

Aichnerc committed:
Implement Product

The product component has been implemented.

Aichnerc committed:
Improve Shop

Both shops now display the new version. The
description is now being queried.

Aichnerc committed:
Improve key handling

A key has been added to the fragments.

Aichnerc committed:
Improve shop

The basic functionality of the shop has been
improved.

Aichnerc committed:
Improve visuals

The css of the shop and its products has been
adjusted. (The featured item badge has been
relocated)

Aichnerc committed:
Improve Product

The product has been updated in its functionality and
design.

Aichnerc committed:
Remove old products

The old products have been removed.

Aichnerc committed:
Remove old shop data

The dummy data of the old shop has been removed.

Aichnerc committed:
Improve Product

The Product component has been improved with
functionality.

Aichnerc committed:
Improve Cart

The cart design has been improved. It is now partly
responsible.

Aichnerc committed:
Improve code quality

An unused text field has been removed.

Aichnerc committed:
Improve code quality

The container and button have been removed from
the Modal component.

Aichnerc committed:
Improve Product

The image height has been limited to 215px.

Aichnerc committed:
Fix Shop products order

The order of the products has been changed.

Aichnerc committed:
Improve code quality

A missing key param has been added.

Aichnerc committed:
Improve LineItem

The design has been improved.

Aichnerc committed:
Improve Cart

The cart has been improved. Unused css has been
removed.

Aichnerc committed:
Fix typo

Renamed Card to Cart

Aichnerc committed:
Fix typo

Card to Cart

Aichnerc committed:
Add availability check

The product variant drop-down options can now be
disabled, if the product is not available.

Aichnerc committed:
Shorten title

The title has been shortened.

Aichnerc committed:
Remove Navbar

The navbar was not used, so it has been removed.

Aichnerc committed:
Implement Cart button

A cart button has been added. It launches the cart.

Aichnerc committed:
Improve code quality

Some unused imports have been removed.
An a tag has been redone as a span tag, not to draw
attention to the required href.

Aichnerc committed:
Add Blackwolf section

The blackwolf section has been added.

Aichnerc committed:
Improve Sub section

The sub section has been improved.

Aichnerc committed:
Improve FAQ

The background of the FAQ section has been changed
to white.

Aichnerc committed:
Improve Product

Sold out products are marked as "Ausverkauft" in
the variant dropdown menu.

Aichnerc committed:
Improve Features section

The Features section has been improved.

Aichnerc committed:
Improve Backwolf section

The Blackwolf section has been improved.

Aichnerc committed:
Improve About section

The center image has been improved.

Aichnerc committed:
Improve Product component

The Product component has been improved.

Aichnerc committed:
Improve Blackwolf

The Blackwolf component has been improved. A
checkout product card has been added.

Aichnerc committed:
Improve Blackwolf

The wolf animation has been adjusted.

Aichnerc committed:
Update version

The version has been updated from 0.2.0 to 0.3.0.

Aichnerc committed:
Remove old Cart content

The commented out old cart code has been removed to
improve code quality.

Aichnerc reviewed (commented):
Looks good to me.

Aichnerc committed:
Improve code quality

The code quality has been improved due to a report by
Codacy.

Aichnerc committed:
Improve code quality

Enforcement of the max. char per line policy.
ES6 fix: Object-Shorthand

schettn commented on src/components/molecules/Cart/index.jsx:
```diff
@@ -0,0 +1,102 @@
+
+class Cart extends React.Component {
+  constructor(props) {
+  super(props);
```
The indents do not look right. Please correct them.

kleberbaum replied:
There should be some kind of indentation.

schettn reviewed (changes requested):
Should be ok to merge after one change.

Aichnerc committed:
Improve code quality

An missing indent has been added because of a
request by @schettn.

schettn reviewed (approved):
Should be ok to merge.

Aichnerc committed:
Improve code quality

A Codacy issue has been fixed. (object-shorthand)

kleberbaum commented on src/components/organisms/Sections/Shop/index.jsx:
```diff
@@ -6,107 +6,116 @@ import React from 'react';
+  }
+
+  render(){
+
```
Please strip all unnecessary newline.

kleberbaum commented on src/components/organisms/Sections/BlackWolf/index.jsx:
```diff
@@ -0,0 +1,139 @@
+  };
+
+  render(){
+
```
Please strip all unnecessary newline.

kleberbaum commented on src/utilities/checkout.js:
```diff
@@ -0,0 +1,175 @@
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
+ */
```
Please end every source file with a newline.

kleberbaum reviewed (changes requested):
Should be ok to merge after a few changes.

Aichnerc committed:
Improve code quality

Some changes regarding code review have been made
due to a request by @kleberbaum.

kleberbaum reviewed (approved):
Should be ok to merge.
